scripts/modules/b2b-account/quotes.js

```diff
@@ -75,16 +80,79 @@ define([
                     });
                 }
             }
+            if (!self.model.get("b2bAccounts")) {
+                var b2bAccount = new B2BAccountModels.b2bAccounts({ pageSize: 200 });
+                b2bAccount.apiGet().then(function (accounts) {
+                    self.model.set("b2bAccounts", accounts);
+                    self.render();
+                });
+            }
+            
+            $('[data-mz-action="applyfilter"]').on('keyup input', function(e) {
+                e.preventDefault();
+                clearTimeout(timeout);
+                FILTERSTRING = "";
+                var dateValue ="";
+                var nameValue = $(this).val();
+                if ($("#expirationdate").val()!=="")
```

Comment:
Fix the formatting here:
`var dateValue ="";` should be `var dateValue = "";`
`if ($("#expirationdate").val()!=="")` should be `if ($("#expirationdate").val() !== "")`

I also don't think you need this if statement at all. Just do this:
```
var nameValue = $(this).val();
var dateValue  = $("#expirationdate").val();
```

It would be even better if the filtering was added to the Backbone model here as two properties. One for the Name filter and one for the Date filter value. Then you don't need to grab the date when name is getting updated and pass it to `filterGrid`. Much cleaner.

Reply:
Done

Reply:
done